Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py

```diff
@@ -0,0 +1,5356 @@
+    """
+    return CommandResults(
+        readable_output=tableToMarkdown(
+            name=readable_message,
+            t=readable_outputs if readable_outputs else outputs,
+            headers=headers,
+            headerTransform=string_to_table_header,
+            removeNull=True,
+        ),
+        outputs_prefix=f"{INTEGRATION_ENTRY_CONTEXT}.{prefix}",
+        outputs_key_field=key_field,
+        outputs=outputs,
+        raw_response=raw_response,
+    )
+
+
+def manual_pagination(response: List[Any], args: dict[str, Any]) -> List[Any]:
+    """
+    Executing Manual paginate_results (using the limit argument).
+
+    Args:
+        response (List[dict[str, Any]]): API response.
+        args (dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        List[dict[str, Any]]: Paginated results.
+    """
+
+    if argToBoolean(args.get("all_results", False)):
+        return response
+
+    limit = arg_to_number(args.get("limit", 50))
+
+    if limit and limit < 1:
+        raise ValueError(ReadableErrors.LIMIT.value)
+
+    return response[:limit]
+
+
+def list_parser(
+    values: List[dict[str, Any]], mapper_command: Callable
+) -> List[dict[str, Any]]:
+    """
+    Handle with parse list of objects.
+
+    Args:
+        values (List[dict[str, Any]]): List of objects to map.
+        mapper_command (Callable): The object parse command.
+
+    Returns:
+        List[dict[str, Any]]: Parsed list.
+    """
+    return [mapper_command(obj) for obj in values]
+
+
+def auto_pagination(
+    request_command: Callable,
+    offset_path: List[str],
+    limit: int | None,
+    page: int | None,
+    page_size: int | None,
+    offset: str | None = None,
+    **kwargs,
+) -> dict[str, Any]:
+    """
+    Handle with pagination when the API supports pagination.
+
+    Args:
+        request_command (Callable): List request command.
+        offset_path (List[str]): The path to the offset arg (in order to handle with page, page_size).
+        limit (int): Limit for pagination.
+        page (int | None): Page for pagination
+        page_size (int | None): Page size for pagination
+        offset (str | None, optional): Offset for pagination. Defaults to None.
+
+    Raises:
+        ValueError: Error with pagination when there is no offset arg.
+
+    Returns:
+        dict[str, Any]: Paginated response.
+    """
+    command_args = copy.deepcopy(kwargs)
+    command_args["offset"] = offset
+    if page and page_size:
+        if page == 1:
+            command_args["limit"] = page_size
+            return request_command(**command_args)
+        calculate = (page - 1) * page_size
+        command_args["limit"] = (
+            calculate
+            if calculate < API_MAX_LIMIT
+            else calculate - calculate % page_size
+        )
+        offset = dict_safe_get(request_command(**command_args), offset_path)
+        if calculate > API_MAX_LIMIT:
```

Comment:
In the API, the limit of list request items is 1000. If the user wants to get objects in 1000 and above indexes, I should run the request a few times and use the offset argument in the API response each time. That is the reason request_command appears more once.
The calculate argument calculates the limit for the API - for example, 500 if the user asks for 500 items, 1000 if the user asks for 1500 items (and the next time it will be the next 500 using the last offset).
I hope it is clear. If not, we can speak about it ;)

Reply:
overall I understand, still need some clarifications tho. in addition - we usually don't allow returning a large number of records in a single command execution (i.e. we set a maximum value of ~200). let's talk next week